src/MaaCore/Assistant.cpp

```diff
@@ -486,6 +492,43 @@ void Assistant::msg_proc()
     }
 }
 
+void Assistant::guard_proc()
+{
+    LogTraceFunction;
+
+    while (true) {
+        std::unique_lock<std::mutex> lock(m_guard_mutex);
+        if (m_thread_exit) {
+            return;
+        }
+
+        if (m_thread_idle || m_tasks_list.empty()) {
+            m_guard_condvar.wait(lock);
+            continue;
+        }
+
+        if (m_ctrler) {
+            const auto& _task = m_tasks_list.front().second;
+            if (auto task = std::dynamic_pointer_cast<StartUpTask>(_task); task) {
+                continue;
+            }
+            const auto& activities = m_ctrler->get_activities();
+            if (!activities) {
+            }
+            else if (activities->find("com.hypergryph.arknights") == std::string::npos) {
+                Log.warn("Assistant::guard_proc | activities died");
+                m_ctrler->start_game("Official");
+                std::this_thread::sleep_for(std::chrono::seconds(120));
+            }
```

Comment:
Isn't this too hardcoded? https://github.com/MaaAssistantArknights/MaaAssistantArknights/blob/29ca30cb99e27814aeae8a0635b99a4eeee75c0f/resource/config.json#L57-L64

Reply:
It's pending. Core needn't be given a client type before

Reply:
`[Adb] -s [AdbSerial] shell \"dumpsys activity top | grep ACTIVITY | grep -i arknights\"`

This should actually work, as it just returns something if arknights is found, otherwise nothing will be returned.